registry_module.go

```diff
@@ -638,14 +685,58 @@ func (r *registryModules) DeleteVersion(ctx context.Context, moduleID RegistryMo
 		return ErrInvalidVersion
 	}
 
+	var Namespace string
+
+	if moduleID.RegistryName == PrivateRegistry {
+		Namespace = moduleID.Organization
+	}
+
+	if moduleID.RegistryName == PublicRegistry {
+		Namespace = moduleID.Namespace
+	}
+
 	u := fmt.Sprintf(
-		"registry-modules/actions/delete/%s/%s/%s/%s",
+		"organizations/%s/registry-modules/%s/%s/%s/%s/%s",
 		url.QueryEscape(moduleID.Organization),
+		url.QueryEscape(string(moduleID.RegistryName)),
+		url.QueryEscape(Namespace),
 		url.QueryEscape(moduleID.Name),
 		url.QueryEscape(moduleID.Provider),
 		url.QueryEscape(version),
 	)
-	req, err := r.client.NewRequest("POST", u, nil)
+	req, err := r.client.NewRequest("DELETE", u, nil)
+	if err != nil && errors.Is(err, ErrResourceNotFound) {
+		return r.deprecatedDeleteVersion(ctx, moduleID, version)
+	}
+
+	return req.Do(ctx, nil)
+}
+
+// DeleteByName is used to delete the entire registry module
+func (r *registryModules) DeleteByName(ctx context.Context, module RegistryModuleDeleteByName) error {
+	if err := module.valid(); err != nil {
+		return err
+	}
+
+	var Namespace string
+
+	if module.RegistryName == PrivateRegistry {
+		Namespace = module.Organization
+	}
+
+	if module.RegistryName == PublicRegistry {
+		Namespace = module.Namespace
+	}
+
+	u := fmt.Sprintf(
+		"organizations/%s/registry-modules/%s/%s/%s",
+		url.QueryEscape(module.Organization),
+		url.QueryEscape(string(module.RegistryName)),
+		url.QueryEscape(Namespace),
+		url.QueryEscape(module.Name),
+	)
+
+	req, err := r.client.NewRequest("DELETE", u, nil)
```

Comment:
We should call the original r.Delete as a fallback for 404s here (until it is removed)